src/osd/OSD.cc

```diff
@@ -8262,6 +8262,19 @@ void OSD::handle_osd_map(MOSDMap *m)
 	  << dendl;
 
   logger->inc(l_osd_map);
+  // since the map received counter is being incremented we know that the osd has recieved a map
+  // first check if it contains a full map in which case increase that logger count by 1 
+  if(!m->maps.empty()){
```

Comment:
Preface: I'm going to comment on some formatting nits.  It's surprisingly disruptive in a large code base for there to be differences in formatting, so it pays to ensure that your changes match the surrounding code as closely as possible.

In the ceph code base generally, it's

```if (condition) {```

Note the spaces between if and ( and between ) and {.  By contrast, no spaces between the () and their contents.  This should be

```if (!m->maps.empty()) {```